src/main/java/com/hierynomus/protocol/commons/buffer/Buffer.java

```diff
@@ -680,7 +680,7 @@ public InputStream asInputStream() {
             @Override
             public int read() throws IOException {
                 try {
-                    return Buffer.this.readByte();
+                    return Buffer.this.readByte() & 0xFF;
```

Comment:
The second test in  src/test/groovy/com/hierynomus/spnego/NegTokenInitTest.groovy covers this.